pkg/openstack/loadbalancer_test.go

```diff
@@ -1108,3 +1108,135 @@ func Test_getStringFromServiceAnnotation(t *testing.T) {
 		})
 	}
 }
+
+func TestBuildBatchUpdateMemberOpts(t *testing.T) {
+	// Sample ServicePort
+	port := corev1.ServicePort{
+		NodePort: 8080,
+	}
+
+	// Sample Nodes
+	node1 := &corev1.Node{
+		ObjectMeta: v1.ObjectMeta{
+			Name: "node-1",
+		},
+		Status: corev1.NodeStatus{
+			Addresses: []corev1.NodeAddress{
+				{
+					Type:    corev1.NodeInternalIP,
+					Address: "192.168.1.1",
+				},
+			},
+		},
+	}
+	node2 := &corev1.Node{
+		ObjectMeta: v1.ObjectMeta{
+			Name: "node-2",
+		},
+		Status: corev1.NodeStatus{
+			Addresses: []corev1.NodeAddress{
+				{
+					Type:    corev1.NodeInternalIP,
+					Address: "192.168.1.2",
+				},
+			},
+		},
+	}
+
+	// Sample serviceConfig
+	svcConf := &serviceConfig{
+		preferredIPFamily:   corev1.IPv4Protocol,
+		lbMemberSubnetID:    "subnet-12345-test",
+		healthCheckNodePort: 8081,
+	}
+
+	// Instantiate an object of your LbaasV2 struct
+	lbaas := &LbaasV2{}
+
+	testCases := []struct {
+		name                 string
+		nodes                []*corev1.Node
+		port                 corev1.ServicePort
+		svcConf              *serviceConfig
+		expectedLen          int
+		expectError          bool
+		checkNewMembersEmpty bool
+	}{
+		{
+			name:        "NodePortequalszero",
+			nodes:       []*corev1.Node{node1, node2},
+			port:        corev1.ServicePort{NodePort: 0},
+			svcConf:     svcConf,
+			expectedLen: 0,
+			expectError: false,
+		},
+		{
+			name:        "ErrNoAddressFound happens for all nodes",
+			nodes:       []*corev1.Node{},
+			port:        port,
+			svcConf:     svcConf,
+			expectedLen: 0,
+			expectError: false,
+		},
+		{
+			name:  "Valid nodes, canUseHTTPMonitor=false",
+			nodes: []*corev1.Node{node1, node2},
+			port:  corev1.ServicePort{NodePort: 8080},
```

Comment:
If I'm not mistaken, this port is identical to `port` (line 1114). Please always use the variable, or always use the literal.